pkg/collector/corechecks/snmp/discovery/discovery.go

```diff
@@ -0,0 +1,294 @@
+package discovery
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/persistentcache"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/snmp/checkconfig"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/snmp/devicecheck"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/snmp/session"
+)
+
+const cacheKeyPrefix = "snmp_corecheck"
+const sysObjectIDOid = "1.3.6.1.2.1.1.2.0"
+
+// Discovery handles snmp discovery states
+type Discovery struct {
+	config            *checkconfig.CheckConfig
+	stop              chan struct{}
+	discDevMu         sync.RWMutex
+	discoveredDevices map[string]Device
```

Comment:
is this `discoveredDevicesByName` ?

Reply:
I changed device digest to be a type, that will help readability. See b88109336